Compfiles/Imo1987P6.lean

```diff
@@ -0,0 +1,271 @@
+/-
+Copyright (c) 2023 The Compfiles Contributors. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jia-Jun Ma
+-/
+import Mathlib.Data.Nat.Prime.Basic
+import Mathlib.Algebra.Order.Floor
+import Mathlib.Data.Real.Sqrt
+import Mathlib.Tactic.Ring
+import Mathlib.Tactic.linarith
+
+import ProblemExtraction
+
+problem_file { tags := [.NumberTheory] }
+
+/-!
+# International Mathematical Olympiad 1987, Problem 6
+Let $n$ be an integer greater than or equal to 2. Prove that if $k^2 + k + n$ is prime for all integers $k$ such that $0 <= k <= \sqrt{n/3}$, then $k^2 + k + n$ is prime for all integers $k$ such that $0 <= k <= n - 2$.
+-/
+
+namespace Imo1987P6
+open Nat
+
+lemma minFac_le_sq {n: ℕ} (hnezero : n≠ 0) (hn : minFac n ≠ n)
```

Comment:
Please put all of these helper lemmas inside a `snip begin ... snip end`.
(See https://github.com/dwrensha/compfiles?tab=readme-ov-file#extracting-problems )